tests/containers/apparmor.pm

```diff
@@ -0,0 +1,57 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Package: apparmor-utils
+# Summary: Test apparmor utilities
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use services::apparmor;
+use utils;
+
+sub is_enabled {
+    # according to Linux' Documentation/admin-guide/LSM/index.rst,
+    # the current security modules can be found in /sys/kernel/security/lsm
+    return script_run 'grep apparmor /sys/kernel/security/lsm' == 0;
+}
+
+sub assert_running {
+    services::apparmor::check_service();
+    services::apparmor::check_aa_status();
+}
+
+sub assert_functional {
+    my $dnsmasq_file = '/etc/apparmor.d/usr.sbin.dnsmasq';
+    if (script_run "test -f $dnsmasq_file") {
+        record_info('Skipped', "File $dnsmasq_file not found, skipping assert_functional test...");
+        return 1;
```

Comment:
I wouldn't skip this one, it is nice, and we mock it anyway, so just make sure that the mock file is cleaned up

Reply:
So, as suggested by @ricardobranco777 this test should not make any writes to the file system. Instead, I checked that a simple service restart [should be enough](https://openqa.suse.de/tests/14985453#step/apparmor/44) to catch any syntax error in /etc/apparmor.d/*.